packages/vm/lib/runTx.ts

```diff
@@ -123,6 +142,10 @@ export default async function runTx(this: VM, opts: RunTxOpts): Promise<RunTxRes
     const result = await _runTx.bind(this)(opts)
     await state.commit()
     debug(`tx checkpoint committed`)
+    if (this._common.isActivatedEIP(2929) && opts.reportAccessList) {
+      result.accessList = state.clearWarmedAccounts(true)
```

Comment:
I know this is WIP, but this currently only supports the accessed items for the root transaction right? It will not report access lists if you `*CALL*` into another address.

Reply:
Hmm it might actually report it, but it will not report if we `REVERT` one of these inner calls, since that removes the accessed items from warmed storage.

Reply:
Hmm, does this happen very often in practice and do you think it's an important miss?

Reply:
(I can add extra logic for it though)

Reply:
Well, internal reverts sometimes happen, and if you don't put these storage slots on the access lists then you are going to get charged to touch a cold account multiple times. So if we want to generate "correct" access lists (we want to be friendly and provide the complete list of the slots and accounts we are going to add in the access list), then we should also include the reverted slots. 

This is actually a bit mind-boggling... There is an edge case here. What if an internal call reverts, but if we add it to the access list then it does not revert? This might also change the behavior of the call; we might now touch other slots. I think we *could* try to rebuild the access lists based upon whatever we just touched, but I think we might get in a flip/flop situation where each time you edit the access list, you get another access list which we already considered before 🤔  (Hope this reasoning makes sense)

Reply:
I will ask on R&D

Reply:
Checkout geth's implementation; https://github.com/ethereum/go-ethereum/pull/22550

They cap the count of attempts to generate an access list at 10.